base/ca/src/main/java/com/netscape/cms/servlet/common/CMCOutputTemplate.java

```diff
@@ -483,6 +531,16 @@ public void createFullResponse(HttpServletResponse resp, Request[] reqs,
 
             resp.setContentType("application/pkcs7-mime");
             resp.setContentLength(contentBytes.length);
+            // Add CMC status as HTTP header for EST (RFC 7030/8951 HTTP status mapping)
+            // CMC status values: SUCCESS=0, FAILED=2, PENDING=3, NO_SUPPORT=4,
+            //                    CONFIRM_REQUIRED=5, POP_REQUIRED=6, PARTIAL=7
+            // EST will map these to HTTP status codes (preliminary implementation):
+            //   SUCCESS -> HTTP 200 OK
+            //   FAILED -> HTTP 400 Bad Request
+            //   NO_SUPPORT/PENDING/CONFIRM_REQUIRED/POP_REQUIRED/PARTIAL -> HTTP 501 Not Implemented
+            if (isESTRequest()) {
+                resp.setHeader("pki-cmc-status", String.valueOf(overallStatus));
```

Comment:
It would be nicer if we could store the string value of the status (e.g. `success`) instead of the integer value as a string (e.g. `0`) since it can make it easier to troubleshoot, but that's not mandatory.